llama_index/finetuning/rerankers/dataset_gen.py

```diff
@@ -0,0 +1,135 @@
+import json
+import random
+from dataclasses import asdict, dataclass
+from typing import Any, List, Optional, Tuple
+
+from llama_index.finetuning import EmbeddingQAFinetuneDataset
+
+
+@dataclass
+class CohereRerankerFinetuneDataset:
+    """Class for keeping track of CohereAI Reranker finetuning training/ validation Dataset."""
+
+    query: str
+    relevant_passages: list
+    hard_negatives: list
```

Comment:
Can we had more descriptive typing here? I think think relavant_passages and hard_negatives should be `List[str]`